### PR TITLE
fix(testnet): look back up to 10,000 blocks on testnet for a legacy chain

### PR DESCRIPTION
**Pull request by teor2345**
## Motivation

I'm seeing one of my testnet instances relaunch with a legacy chain error.

## Solution

- Increase the maximum number of legacy chain blocks to 10,000
- Make the tests use 100 blocks as the limit

## Review

Anyone can review this PR, it's not urgent.

### Reviewer Checklist

  - [ ] CI passes

### CODE REVIEW

**Comment by teor2345** [comment]
@mergifyio update

File: zebra-state/src/service/check.rs [hunk 1]
```diff
@@ -316,7 +321,7 @@ where
 
         // If we are past our NU5 activation height, but there are no V5 transactions in recent blocks,
         // the Zebra instance that verified those blocks had no NU5 activation height.
```
Comment (upbqdn) [inline zebra-state/src/service/check.rs]: Shouldn't the `Zebra instance` be `other peer`?
```suggestion
        // the other peer that verified those blocks had no NU5 activation height.
```

**Review by upbqdn — DISMISSED** [review]
All looks good. I left one small note.

File: zebra-state/src/service/check.rs (hunk 1 above)
Comment (teor2345) [reply zebra-state/src/service/check.rs]: We're checking the local cached state here, without any network connections:
```suggestion
        // the last Zebra instance that updated this cached state had no NU5 activation height.
```

**Review by upbqdn — DISMISSED** [review]

**Comment by teor2345** [comment]
I needed to merge `use` changes from `main` and this PR, then do a rustfmt using `cargo fmt --all`.

**Comment by teor2345** [comment]
Failed due to #5069:
https://github.com/ZcashFoundation/zebra/actions/runs/3087790499/jobs/4994029955

**Comment by teor2345** [comment]
This is one I haven't seen before:
> - Internal error. Please try again or contact Google Support. (Code: '5E917363FAD65.6B58463.F6090D12')
> - Internal error. Please try again or contact Google Support. (Code: '5E917363FB573.6B58463.F6090D12')

https://github.com/ZcashFoundation/zebra/actions/runs/3087790499/jobs/4994831429#step:6:131

**Review by upbqdn — APPROVED** [review]

**Pull request merged**